### ISSUE TITLE
Stall on Generating Samples

### ISSUE THREAD

**Issue by JousterL** [d8ahazard/sd_dreambooth_extension#909]
### Kindly read the entire form below and fill it out with the requested information.

**Please find the following lines in the console and paste them below. _If you do not provide this information, your
issue will be automatically closed._**

`
Python 3.10.9 (tags/v3.10.9:1dd9be6, Dec  6 2022, 20:01:21) [MSC v.1934 64 bit (AMD64)]
Commit hash: 3e0f9a75438fa815429b5530261bcf7d80f3f101
Installing requirements for Web UI

#######################################################################################################
Initializing Dreambooth
If submitting an issue on github, please provide the below text for debugging purposes:

Python revision: 3.10.9 (tags/v3.10.9:1dd9be6, Dec  6 2022, 20:01:21) [MSC v.1934 64 bit (AMD64)]
Dreambooth revision: 21ddebabb47274255d516e655f997d9ddadc0fe4
SD-WebUI revision: 3e0f9a75438fa815429b5530261bcf7d80f3f101

Checking Dreambooth requirements...
[+] bitsandbytes version 0.35.0 installed.
[+] diffusers version 0.10.2 installed.
[+] transformers version 4.25.1 installed.
[+] xformers version 0.0.14.dev0 installed.
[+] torch version 1.13.1+cu117 installed.
[+] torchvision version 0.14.1+cu117 installed.
`

**Have you read the [Readme](https://github.com/d8ahazard/sd_dreambooth_extension#readme)?**
Yes
**Have you completely restarted the stable-diffusion-webUI, not just reloaded the UI?**
Yes
**Have you updated Dreambooth to the latest revision?**
Yes
**Have you updated the Stable-Diffusion-WebUI to the latest version?**
Yes
**No, really. Please save us both some trouble and update the SD-WebUI and Extension and restart before posting this.
Reply 'OK' Below to acknowledge that you did this.**
OK
**Describe the bug**

I was originally using an older (mid-December) commit for Dreambooth, and I experienced no significant bottlenecks doing a training run. With the current revision, I have noticed normal behavior for the actual training run, but then it 'hangs' on "Generating Samples" at 33%, and Python goes down to a single CPU core in utilization. I see no other utilization (GPU, disk, etc) during this time. This persists for at least 25 minutes, and obviously significantly slows down the training process (what used to take ~12-18 hours is now a multi-day effort).

**Environment**

What OS?
Windows 11
If Windows - WSL or native?
Native
What GPU are you using?
MSI GeForce RTX 4090 Suprim Liquid X 24G
**Screenshots/Config**
![image](https://user-images.githubusercontent.com/439681/217126502-f163efae-3ea7-4d90-a0be-02f071d63db7.png)

**Comment by Arilziem** [comment]
I am seeing this as well (on dev branch [73fbcb7](https://github.com/d8ahazard/sd_dreambooth_extension/commit/73fbcb7ff1d25168b9c389522f8e131c313430ae)), although it takes a few minutes for me rather than 25. This is happening after the sample image inference, when the statistics are being plotted (as the 33% would suggest).

I did not pay specific attention to it, but in retrospect it seemed to be taking longer with more training steps (also after interrupting and resuming training).

**Comment by minienglish1** [comment]
I noticed the exact same problem, and also suspected it was the graph taking too long to draw.  Especially troublesome when doing long trainings.

Does disabling samples also disable the loss graph?

**Comment by Arilziem** [comment]
> Does disabling samples also disable the loss graph?

Yes.

**Comment by d8ahazard** [comment]
This should be resolved with https://github.com/d8ahazard/sd_dreambooth_extension/releases/tag/1.0.11

Feel free to open a new issue if you're still having problems.

**Comment by orenong** [comment]
Still have this issue

**Comment by Arilziem** [comment]
@orenong See #1035 (still an issue though)
